### PR TITLE
[QI2-1096] pennylane device implementation

### PR DESCRIPTION
**Pull request by napocornejo**

### CODE REVIEW

File: tests/test_device.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    device = QI2Device(backend=QI2_backend)
+
+    @qml.qnode(device=device)
+    # untyped decorator in pennylane, so mypy will complain here
+    def quantum_function():  # type: ignore
+        qml.Hadamard(wires=[0])
+        return qml.expval(qml.PauliX(wires=[0]))
+
+    mocker.patch("qiskit_quantuminspire.qi_backend.QIJob")
+    assert quantum_function() is not None
```
Comment (NischalQuTech) [inline tests/test_device.py:26]:
```suggestion
    mock_job = MagicMock()
    mock_submit = MagicMock()
    mock_job.submit = mock_submit

    mocker.patch("qiskit_quantuminspire.qi_backend.QIJob", return_value=mock_job)
    assert quantum_function() is not None
    mock_job.submit.assert_called()
```

**Review by NischalQuTech — CHANGES_REQUESTED** [review]

**Comment by NischalQuTech** [comment]
I think we need a method to list all the backends for the user and get a backend by name. It would be helpful to also show an example workflow in Readme.md file

File: tests/test_device.py (hunk 1 above)
Comment (napocornejo) [reply tests/test_device.py:26]: Done.

File: pennylane_quantuminspire2/qi_device.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+class QI2Device(RemoteDevice):  # type: ignore[misc]
+
+    _qi_provider = QIProvider()
+
+    def __init__(self, backend: QIBackend, **kwargs: Any) -> None:
+        super().__init__(wires=backend.num_qubits, backend=backend, shots=backend.default_shots, **kwargs)
+
+    @classmethod
+    def get_backends(cls) -> Sequence[QIBackend]:
```
Comment (NischalQuTech) [inline pennylane_quantuminspire2/qi_device.py]: shall we name this method as `backends` ? That makes it more consistent

Comment (napocornejo) [reply pennylane_quantuminspire2/qi_device.py]: done

**Review by NischalQuTech — APPROVED** [review]

**Pull request merged**